### ISSUE TITLE
Special UUID-type handling broken in some cases

### ISSUE THREAD

**Issue by jskov-jyskebank-dk** [jskov/openapi-jaxrs-client#553]
In the below API, the RoomId is handled as a DTO whereas UUID is handled as the java UUID type.

The parser logic handling this takes its cue from OpenApi schema type (UUIDSchema) not the type name.
So it is unclear if this is a OpenApi problem, or mada parser jumping the gun somewhere else.

Check if it is because the pattern is missing (could trigger magic in OpenAPI).

```yaml
  /api/uuid/query:
    get:
      tags:
      - Resource Dtos
      parameters:
        - in: query
          name: roomId
          schema:
            $ref: '#/components/schemas/RoomId'
...
    UUID:
      format: uuid
      pattern: "[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}"
      type: string
    RoomId:
      description: UUID of the room.
      example: e017e2ea-ff98-4516-9a56-bbd02e1f161a
      format: uuid
      type: string

```
(MM)

**Comment by jskov-jyskebank-dk** [comment]
Not due to extra properties

**Comment by jskov-jyskebank-dk** [comment]
Fixed by #554
